tests/testScrapPaper.js

```diff
@@ -0,0 +1,386 @@
+const assert = require('chai').assert;
+const fetch = require('node-fetch');
+const cheerio = require('cheerio');
+
+const { readPdf } = require('../lib/scrapPaperReader');
+const { getBarcodeSegments } = require('../lib/scrapPaperMaker');
+const config = require('../lib/config');
+const util = require('util');
+const sqldb = require('../prairielib/lib/sql-db');
+const sqlLoader = require('../prairielib/lib/sql-loader');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+// const FormData = require('form-data');
+
+const { saveOrGrade } = require('./helperClient');
+const helperServer = require('./helperServer');
+
+const jsCrc = require('js-crc');
+
+const testLabel = 'ANY TEST LABEL';
+const testNumPages = 5; // has to be reasonably small for pdf to be converted/decoded quickly in test
+
+/**
+ * Set the active user within Prairie Learn's test environment.
+ * @param {object} user
+ */
+const setUser = (user) => {
+  config.authUid = user.authUid;
+  config.authName = user.authName;
+  config.authUin = user.authUin;
+};
+
+/**
+ * Gets an instance question URL for the current user
+ * @param {string} baseUrl
+ * @param {string} hm1AutomaticTestSuiteUrl
+ * @returns {string}
+ */
+const getBarcodeSubmissionUrl = async (baseUrl, hm1AutomaticTestSuiteUrl) => {
+  const res = await fetch(hm1AutomaticTestSuiteUrl);
+  assert.equal(res.ok, true);
+
+  const hm1Body = await res.text();
+  assert.include(hm1Body, 'HW1.12. Barcode submission');
+
+  const $hm1Body = cheerio.load(hm1Body);
+  return (
+    baseUrl.replace('/pl', '') + $hm1Body('a:contains("HW1.12. Barcode submission")').attr('href')
+  );
+};
+
+const getScrapPaperPayload = ($page, numPages, pageLabel) => {
+  return {
+    num_pages: numPages,
+    page_label: pageLabel,
+    __csrf_token: $page('form > input[name="__csrf_token"]').val(),
+    __action: $page('form > input[name="__action"]').val(),
+  };
+};
+
+// const getScanPaperPayload = ($page, pdfBuffer) => {
+//   const formData = new FormData();
+//   formData.append('file', pdfBuffer, 'ANY FILENAME.pdf');
+//   formData.append('__csrf_token', $page('form > input[name="__csrf_token"]').val());
+//   formData.append('__action', $page('form > input[name="__action"]').val());
+//   return formData;
+// };
+
+describe('Barcode generation, student submission, and scanning process', function () {
+  this.timeout(75000);
```

Comment:
I ran the test suite locally and it only took 5 seconds - this seems a bit excessive.

Reply:
Yeah, it was optimized. 75000 is too long, but the second part takes longer too. Up to a judgment call based on what people's development environments are like. Maybe 30000?